move-mutation-test/src/mutation_test.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright © Eiger
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::TestBuildConfig;
+use anyhow::Error;
+use aptos_framework::extended_checks;
+use aptos_gas_schedule::{MiscGasParameters, NativeGasParameters, LATEST_GAS_FEATURE_VERSION};
+use aptos_types::on_chain_config::{
+    aptos_test_feature_flags_genesis, Features, TimedFeaturesBuilder,
+};
+use aptos_vm::natives;
+use move_cli::base::test::UnitTestResult;
+use move_command_line_common::address::NumericalAddress;
+use move_package::BuildConfig;
+use move_unit_test::UnitTestingConfig;
+use move_vm_runtime::native_functions::NativeFunctionTable;
+use std::path::Path;
+use termcolor::WriteColor;
+
+/// The `run_tests` function is responsible for running the tests for the provided package.
+///
+/// # Arguments
+///
+/// * `cfg` - A `TestBuildConfig` representing the test configuration.
+/// * `package_path` - A `Path` to the package.
+/// * `error_writer` - `&mut dyn std::io::Write` the error writer.
+///
+/// # Returns
+///
+/// * `anyhow::Result<()>` - The result of the test suite for the package.
+pub(crate) fn run_tests<W: WriteColor + Send>(
```

Comment:
We might want to document the aptos-core function from which this is based on, so that eventually we can refactor appropriately to remove code duplication.

Reply:
I left a reference to this `aptos-core/crates/aptos/src/move_tool/mod.rs` - this function is based upon `TestPackage::execute` method mostly/